app/src/test/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStackTest.kt

```diff
@@ -148,64 +140,82 @@ class NgVpnNetworkStackTest {
+        ngVpnNetworkStack.onDnsResolved(createDnsRecord(THIRD_PARTY_TRACKER.tracker.hostname, "1.1.1.1"))
 
-        assertEquals(1, vpnAddressLookupDao.getAll().size)
+        assertTrue(ngVpnNetworkStack.isAddressBlocked(AddressRR("1.1.1.1", uid)))
     }
 
     @Test
-    fun whenOnStopVpnThenNeverPersistDomainsThatAreNotTrackers() {
-        whenever(appTrackerRepository.findTracker(eq("www.duckduckgo.com"), anyString())).thenReturn(AppTrackerType.NotTracker)
-        ngVpnNetworkStack.onSniResolved(SniRR("www.duckduckgo.com", "1.1.1.1"))
+    fun whenIsAddressBlockedAndDomainIsFirstPartyTrackerThenReturnFalse() {
+        val uid = 1200
 
-        ngVpnNetworkStack.onStopVpn()
+        whenever(appTrackerRepository.findTracker(eq(THIRD_PARTY_TRACKER.tracker.hostname), anyString())).thenReturn(FIRST_PARTY_TRACKER)
+        whenever(vpnAppTrackerBlockingDao.getRuleByTrackerDomain(THIRD_PARTY_TRACKER.tracker.hostname)).thenReturn(null)
+        ngVpnNetworkStack.onDnsResolved(createDnsRecord(THIRD_PARTY_TRACKER.tracker.hostname, "1.1.1.1"))
 
-        assertEquals(0, vpnAddressLookupDao.getAll().size)
+        assertFalse(ngVpnNetworkStack.isAddressBlocked(AddressRR("1.1.1.1", uid)))
     }
 
     @Test
-    fun whenOnStopVpnAfterDnsResolvedPersistTrackerLookupCache() {
-        ngVpnNetworkStack.onDnsResolved(DnsRR(0, "api2.branch.com", "api2.branch.com", "1.1.1.1", 0))
+    fun whenIsAddressBlockedAndDomainIsNotTrackerThenReturnFalse() {
+        val uid = 1200
 
-        ngVpnNetworkStack.onStopVpn()
+        whenever(appTrackerRepository.findTracker(eq(THIRD_PARTY_TRACKER.tracker.hostname), anyString())).thenReturn(AppTrackerType.NotTracker)
+        whenever(vpnAppTrackerBlockingDao.getRuleByTrackerDomain(THIRD_PARTY_TRACKER.tracker.hostname)).thenReturn(null)
+        ngVpnNetworkStack.onDnsResolved(createDnsRecord(THIRD_PARTY_TRACKER.tracker.hostname, "1.1.1.1"))
 
-        assertEquals(1, vpnAddressLookupDao.getAll().size)
+        assertFalse(ngVpnNetworkStack.isAddressBlocked(AddressRR("1.1.1.1", uid)))
     }
 
     @Test
-    fun whenIsAddressBlockedAndDomainIsThirdPartyTrackerThenReturnTrue() {
+    fun whenIsDomainBlockedAndDomainIsFirstPartyTrackerThenReturnFalse() {
         val uid = 1200
+        val hostname = TEST_APP_TRACKER.hostname
 
-        whenever(appTrackerRepository.findTracker(eq(THIRD_PARTY_TRACKER.tracker.hostname), anyString())).thenReturn(THIRD_PARTY_TRACKER)
-        whenever(vpnAppTrackerBlockingDao.getRuleByTrackerDomain(THIRD_PARTY_TRACKER.tracker.hostname)).thenReturn(null)
-        ngVpnNetworkStack.onSniResolved(SniRR(THIRD_PARTY_TRACKER.tracker.hostname, "1.1.1.1"))
+        whenever(appTrackerRepository.findTracker(eq(hostname), anyString())).thenReturn(FIRST_PARTY_TRACKER)
+        whenever(vpnAppTrackerBlockingDao.getRuleByTrackerDomain(hostname)).thenReturn(null)
+        ngVpnNetworkStack.isDomainBlocked(DomainRR(hostname, uid))
```

Comment:
Do we need this extra call to `isDomainBlocked`? I think we should be able to immediately assert that it returns true/false for this test and the ones below.

Reply:
oh, dumb error when I fixing the test. Will remove